app/src/main/java/com/github/gotify/messages/MessagesActivity.kt

```diff
@@ -544,109 +550,74 @@ internal class MessagesActivity :
                 })
     }
 
-    private inner class LoadMore : AsyncTask<Long?, Void?, List<MessageWithImage>>() {
-        override fun doInBackground(vararg appId: Long?): List<MessageWithImage> {
-            return viewModel.messages.loadMore(appId.first()!!)
-        }
-
-        override fun onPostExecute(messageWithImages: List<MessageWithImage>) {
-            updateMessagesAndStopLoading(messageWithImages)
+    private suspend fun loadMore(appId: Long) {
+        val messagesWithImages = viewModel.messages.loadMore(appId)
+        withContext(Dispatchers.Main) {
+            updateMessagesAndStopLoading(messagesWithImages)
```

Comment:
I can't remember why I did this, let me check.

Reply:
Recovered my thoughts:

The function is called twice via the mentioned `withContext(Dispatchers.Main)`, but one call (see [here](https://github.com/cyb3rko/android/blob/cebf7f4afeda8a5b343de719f95d01a5bea750e3/app/src/main/java/com/github/gotify/messages/MessagesActivity.kt#L186-L190)) is directly done on the main thread, that's why the additional call of `withContext(Dispatchers.Main)` in the function would be duplicate.

You decide which way we go though.

Reply:
Generally it is no problem if withContext is executed twitch twice with the same dispatcher, it should just no-op then. But your argument still stands a little different because onUpdateApps isn't a suspend function, thus, we'd need to launch a coroutine just to go back to the main thread.

Let's leave it as is, as I can't think of a better solution right now.